dns_cache/storage.go

```diff
@@ -0,0 +1,81 @@
+package dns_cache
+
+import (
+	"net"
+	"reflect"
+	"time"
+
+	cache "github.com/pmylund/go-cache"
+)
+
+type DnsCacheItem struct {
```

Comment:
a little confusing the purpose of this struct, would be equal to use just `type DnsCacheItem []string` or store in cache just `string` (see my comment about always using 1st element from addrs list above)